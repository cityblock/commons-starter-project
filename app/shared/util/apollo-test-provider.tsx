import { InMemoryCache } from 'apollo-cache-inmemory';
import { ApolloClient } from 'apollo-client';
import { SchemaLink } from 'apollo-link-schema';
import { print } from 'graphql';
import { addMockFunctionsToSchema, makeExecutableSchema } from 'graphql-tools';
import createHistory from 'history/createBrowserHistory';
import * as React from 'react';
import { ApolloProvider } from 'react-apollo';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import * as schema from '../../../server/graphql/schema.graphql';
import ReduxConnectedIntlProvider from '../../redux-connected-intl-provider';
import createStore from '../../store';

const history = createHistory();
const store = createStore(history);

interface IProps {
  graphqlMocks: any;
}

// Helper in our codebase to create the apollo client
// from http://blog.dideric.is/2018/03/18/Testing-apollo-containers/
export default class ApolloTestProvider extends React.Component<IProps> {
  static createSchema = () => makeExecutableSchema({ typeDefs: print(schema) });

  apolloClient: any;
  schema: any;

  constructor(props: IProps) {
    super(props);
    // Every instance should have it's own schema instance so tests
    // don't bleed into one another
    this.schema = ApolloTestProvider.createSchema();
    this.apolloClient = new ApolloClient<any>({
      link: new SchemaLink({ schema: this.schema }),
      cache: new InMemoryCache().restore({}) as any,
    });
    this.addDefaultMocks();
  }

  componentWillMount() {
    const { graphqlMocks } = this.props;
    this.mockGraphql(graphqlMocks);
  }

  componentWillReceiveProps({ graphqlMocks }: any) {
    this.mockGraphql(graphqlMocks);
  }

  addDefaultMocks() {
    addMockFunctionsToSchema({
      schema: this.schema,
    });
  }

  mockGraphql(mocks = {}) {
    addMockFunctionsToSchema({
      schema: this.schema,
      mocks,
    });
  }

  render() {
    const { children } = this.props;
    return (
      <Provider store={store}>
        <ApolloProvider client={this.apolloClient}>
          <ReduxConnectedIntlProvider defaultLocale="en">
            <MemoryRouter>{children}</MemoryRouter>
          </ReduxConnectedIntlProvider>
        </ApolloProvider>
      </Provider>
    );
  }
}
