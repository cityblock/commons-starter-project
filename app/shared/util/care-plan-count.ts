import { FullCarePlanSuggestionFragment } from '../../graphql/types';

export function getConcernCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (!carePlanSuggestions.length) {
    return 0;
  }

  const concernSuggestions = carePlanSuggestions.filter(
    suggestion => suggestion.suggestionType === 'concern',
  );

  return concernSuggestions.length;
}

export function getGoalSuggestions(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (!carePlanSuggestions.length) {
    return [];
  }

  return carePlanSuggestions.filter(suggestion => suggestion.suggestionType === 'goal');
}

export function getGoalCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  return getGoalSuggestions(carePlanSuggestions).length;
}

export function getTaskCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (getGoalCount(carePlanSuggestions) === 0) {
    return 0;
  }

  const goalSuggestions = getGoalSuggestions(carePlanSuggestions);

  const taskSuggestions = goalSuggestions
    .map(goalSuggestion => goalSuggestion.goalSuggestionTemplate!.taskTemplates)
    .reduce((taskSuggestions1, taskSuggestions2) => taskSuggestions1.concat(taskSuggestions2));

  return (taskSuggestions || []).length;
}
