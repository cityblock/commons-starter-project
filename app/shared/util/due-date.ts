export const checkIfDueSoon = (dateString: string | null) => {
  let dueSoon = false;

  if (dateString) {
    const taskDueDate = new Date(dateString);
    dueSoon = taskDueDate.valueOf() < Date.now() + 60 * 60 * 24 * 1000;
  }

  return dueSoon;
};
