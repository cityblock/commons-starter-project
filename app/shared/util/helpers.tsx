export function ToArray(enumme: any) {
  return Object.keys(enumme).map(key => enumme[key]);
}
