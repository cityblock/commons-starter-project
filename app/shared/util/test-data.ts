export const clinic = {
  id: 'clinic-id',
  name: 'Home clinic',
};

export const currentUser = {
  id: 'id',
  locale: 'en',
  phone: '(212) 555-8394',
  firstName: 'first',
  lastName: 'last',
  userRole: 'physician' as any,
  email: 'a@b.com',
  homeClinicId: clinic.id,
  googleProfileImageUrl: null,
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  permissions: 'orange' as any,
};

export const featureFlags = {
  isBuilderEnabled: false,
  isManagerEnabled: false,
  canChangeUserPermissions: false,
  canDeleteUsers: false,
  canBulkAssign: false,
  canEditCareTeam: true,
  canViewAllMembers: true,
  canEditAllMembers: true,
  canViewMembersOnPanel: true,
  canEditMembersOnPanel: true,
  canShowAllMembersInPatientPanel: true,
  canDisenrollPatient: true,
  canAutoBreakGlass: false,
};

export const user = {
  id: 'id2',
  locale: 'en',
  phone: '(212) 555-2828',
  firstName: 'first',
  lastName: 'last',
  userRole: 'physician' as any,
  email: 'b@c.com',
  homeClinicId: clinic.id,
  googleProfileImageUrl: null,
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  permissions: 'blue' as any,
};

export const address1 = {
  id: 'address-1-id',
  street: '55 Washington St',
  city: 'Brooklyn',
  state: 'NY',
  description: 'work',
  zip: '11201',
};

export const address2 = {
  id: 'address-2-id',
  street: '101 Fake St',
  city: 'Cambridge',
  state: 'MA',
  zip: '02139',
};

export const address3 = {
  id: 'address-3-id',
  street: '20 Main St',
  zip: '12345',
  description: 'generic',
};

interface IAddress {
  id: string;
  street: string | null;
  state: string | null;
  zip: string | null;
  city: string | null;
  description: string | null;
}

export const patient = {
  id: 'patient-id',
  firstName: 'Bob',
  middleName: null,
  lastName: 'Smith',
  dateOfBirth: '01/01/1999',
  homeClinicId: clinic.id,
  createdAt: '2017-09-07T13:45:14.532Z',
  scratchPad: 'Note',
  consentToCall: true,
  consentToText: true,
  patientInfo: {
    id: 'patient-info-id',
    gender: 'male',
    language: 'en',
    primaryAddress: address1,
    addresses: [address1, address2, address3] as IAddress[],
  },
  patientDataFlags: [],
};

export const concern = {
  id: 'concern-id',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  dueAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  title: 'Concern Title',
  diagnosisCodes: [],
};

export const riskAreaGroup = {
  id: 'ghost',
  title: "Jon Snow's Direwolf",
  shortTitle: 'Ghost',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  order: 1,
  mediumRiskThreshold: 30,
  highRiskThreshold: 9000,
  riskAreas: [],
};

export const riskArea = {
  id: 'cool-risk-area-id',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  dueAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  title: 'Risk Area Title',
  order: 1,
  mediumRiskThreshold: 5,
  highRiskThreshold: 8,
  assessmentType: 'manual' as any,
  riskAreaGroupId: riskAreaGroup.id,
  questions: [],
  screeningTools: [],
};

export const automatedRiskArea = {
  id: 'ghost',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  dueAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  title: "Jon Snow's Direwolf",
  order: 1,
  mediumRiskThreshold: 5,
  highRiskThreshold: 8,
  assessmentType: 'automated' as any,
  riskAreaGroup,
  questions: [],
};

export const riskAreaAssessmentSubmission = {
  carePlanSuggestions: [],
  completedAt: null,
  createdAt: 'Thu Dec 14 2017 13:06:41 GMT-0500 (EST)',
  deletedAt: null,
  id: 'risk-area-id',
  patientId: patient.id,
  riskAreaId: riskArea.id,
  updatedAt: 'Thu Dec 14 2017 13:06:41 GMT-0500 (EST)',
  userId: user.id,
};

export const answer = {
  id: 'answer-id',
  displayValue: 'answer value',
  value: 'true',
  valueType: 'boolean' as any,
  riskAdjustmentType: 'increment' as any,
  inSummary: true,
  summaryText: 'summary text',
  questionId: 'cool-task-id',
  order: 1,
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  concernSuggestions: [],
  goalSuggestions: [],
  riskArea: null,
  screeningTool: null,
};

export const otherTextAnswer = {
  id: 'other-text-answer-id',
  displayValue: 'other',
  value: 'other',
  valueType: 'string' as any,
  riskAdjustmentType: 'inactive' as any,
  inSummary: false,
  summaryText: null,
  questionId: 'cool-task-id',
  order: 1,
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  concernSuggestions: [],
  goalSuggestions: [],
  riskArea: null,
  screeningTool: null,
};

export const answerMutiSelect = {
  id: '456',
  displayValue: 'Answer',
  value: 'answer',
  valueType: 'string' as any,
  questionId: '123',
  order: 1,
  riskAdjustmentType: null,
  inSummary: false,
  summaryText: null,
  concernSuggestions: [],
  goalSuggestions: [],
  riskArea: null,
  screeningTool: null,
};

export const taskTemplate = {
  id: 'task-template-id',
  title: 'Task Template Title',
  completedWithinNumber: 1,
  completedWithinInterval: 'hour' as any,
  repeating: false,
  goalSuggestionTemplateId: 'goal-suggestion-template-id',
  priority: 'high' as any,
  careTeamAssigneeRole: 'physician' as any,
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  CBOCategoryId: null,
};

export const goalSuggestionTemplate = {
  id: 'goal-suggestion-template-id',
  title: 'Goal Suggestion Template Title',
  taskTemplates: [taskTemplate],
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
};

export const answerWithConcernAndGoal = {
  ...answer,
  concernSuggestions: [concern],
  goalSuggestions: [goalSuggestionTemplate],
};

export const goal = {
  id: 'goal-id',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  title: 'Goal Title',
  taskTemplates: [taskTemplate],
};

export const question = {
  id: 'cool-question-id',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  dueAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  title: 'Question Title',
  order: 1,
  validatedSource: 'validated source',
  answerType: 'radio' as any,
  answers: [answer],
  riskAreaId: 'risk-area-id',
  screeningToolId: null,
  applicableIfType: 'allTrue' as any,
  applicableIfQuestionConditions: [
    {
      id: 'question-condition',
      questionId: 'cool-question-id',
      answerId: 'answer-id',
    },
  ],
  computedFieldId: null,
  computedField: null,
  otherTextAnswerId: null,
};

export const questionWithOtherTextAnswer = {
  id: 'cool-question-id',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  dueAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  title: 'Question Title',
  order: 1,
  validatedSource: 'validated source',
  answerType: 'radio' as any,
  answers: [otherTextAnswer, answer],
  riskAreaId: 'risk-area-id',
  screeningToolId: null,
  applicableIfType: 'allTrue' as any,
  applicableIfQuestionConditions: [
    {
      id: 'question-condition',
      questionId: 'cool-question-id',
      answerId: 'answer-id',
    },
  ],
  computedFieldId: null,
  computedField: null,
  otherTextAnswerId: otherTextAnswer.id,
};

export const questionWithAnswerWithConcernAndGoal = {
  id: 'different-question-id',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  dueAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  title: 'different Question Title',
  order: 1,
  validatedSource: 'validated source',
  answerType: 'radio' as any,
  answers: [answerWithConcernAndGoal],
  riskAreaId: 'risk-area-id',
  screeningToolId: null,
  applicableIfType: 'allTrue' as any,
  applicableIfQuestionConditions: [
    {
      id: 'question-condition',
      questionId: 'cool-question-id',
      answerId: 'answer-id',
    },
  ],
};

export const carePlanSuggestionWithConcern = {
  id: 'care-plan-suggestion-id',
  patientId: 'patient-id',
  patient,
  suggestionType: 'concern' as any,
  concernId: 'concern-id',
  concern,
  goalSuggestionTemplateId: null,
  goalSuggestionTemplate: null,
  acceptedById: null,
  acceptedBy: null,
  dismissedById: null,
  dismissedBy: null,
  dismissedReason: null,
  dismissedAt: null,
  acceptedAt: null,
  createdAt: '2017-08-16T19:27:36.378Z',
  updatedAt: '2017-08-16T19:27:36.378Z',
  patientScreeningToolSubmissionId: null,
};

export const carePlanSuggestionWithGoal = {
  id: 'care-plan-suggestion-id',
  patientId: 'patient-id',
  patient,
  suggestionType: 'goal' as any,
  concernId: null,
  concern: null,
  goalSuggestionTemplateId: 'goal-suggestion-template-id',
  goalSuggestionTemplate: {
    id: 'goal-suggestion-template-id',
    title: 'Goal Title',
    taskTemplates: [
      {
        id: 'task-template-1',
        title: 'Task Title',
        priority: 'high' as any,
        completedWithinNumber: 1,
        completedWithinInterval: 'week' as any,
        repeating: false,
        goalSuggestionTemplateId: 'goal-suggestion-template-1',
        careTeamAssigneeRole: 'physician' as any,
        createdAt: '2017-08-16T19:27:36.378Z',
        updatedAt: '2017-08-16T19:27:36.378Z',
        deletedAt: null,
        CBOCategoryId: null,
      },
    ],
    createdAt: '2017-08-16T19:27:36.378Z',
    updatedAt: '2017-08-16T19:27:36.378Z',
    deletedAt: null,
  },
  acceptedById: null,
  acceptedBy: null,
  dismissedById: null,
  dismissedBy: null,
  dismissedReason: null,
  dismissedAt: null,
  acceptedAt: null,
  createdAt: '2017-08-16T19:27:36.378Z',
  updatedAt: '2017-08-16T19:27:36.378Z',
  patientScreeningToolSubmissionId: null,
};

export const task = {
  id: 'task-id',
  title: 'Task Title',
  description: null,
  patientId: 'patient-id',
  patient,
  dueAt: '2017-09-07T13:45:14.532Z',
  priority: 'high' as any,
  createdBy: currentUser,
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  deletedAt: null,
  completedBy: null,
  completedAt: null,
  assignedTo: null,
  followers: [],
  patientGoal: null,
  CBOReferralId: null,
  CBOReferral: null,
  assignedToId: 'jonSnow',
};

export const comment = {
  id: 'comment-id',
  body: 'body',
  user: currentUser,
  taskId: task.id,
  createdAt: 'Thu Jul 14 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 14 2017 16:52:56 GMT-0400 (EDT)',
};

export const patientConcernForGoal = {
  id: 'nightKingMarchingSouth',
  order: 1,
  concernId: 'concern-id',
  concern,
  patientGoals: [],
  patientId: 'patient-id',
  patient,
  startedAt: null,
  completedAt: null,
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  deletedAt: null,
};

export const patientGoal = {
  id: 'patient-goal-id',
  title: 'Goal Title',
  patientId: 'patient-id',
  patient,
  patientConcernId: 'nightKingMarchingSouth',
  patientConcern: patientConcernForGoal,
  goalSuggestionTemplateId: 'goal-suggestion-template-id',
  goalSuggestionTemplate,
  tasks: [task],
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  deletedAt: null,
};

export const assignedTask = {
  ...task,
  patient,
  patientId: patient.id,
  assignedTo: currentUser,
  createdBy: currentUser,
  followers: [currentUser],
};

export const completedTask = {
  ...task,
  patientId: patient.id,
  createdById: currentUser.id,
  assignedToId: currentUser.id,
  completedById: currentUser.id,
};

export const patientConcern = {
  id: 'patient-concern-id',
  order: 1,
  concernId: 'concern-id',
  concern,
  patientGoals: [patientGoal],
  patientId: 'patient-id',
  patient,
  startedAt: null,
  completedAt: null,
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  deletedAt: null,
};

export const patientAnswer = {
  id: 'patient-answer',
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  deletedAt: null,
  answerId: answer.id,
  answerValue: answer.value,
  patientId: patient.id,
  applicable: true,
  patientScreeningToolSubmissionId: null,
  question: {
    id: question.id,
    title: question.title,
  },
  answer,
};

export const encounter = {
  encounterType: 'encounter type',
  providerName: 'provider name',
  providerRole: 'provider role',
  location: 'location',
  diagnoses: [
    {
      code: 'code',
      codeSystem: 'code system',
      description: 'desc',
    },
  ],
  reasons: ['reason'],
  dateTime: '10/10/2010',
};

export const medication = {
  id: 'med id',
  name: 'name',
  dosageInstructions: 'once daily',
};

export const eventNotification = {
  id: 'event-notificaiton-id',
  title: 'event notification',
  userId: currentUser.id,
  user: currentUser,
  taskEventId: 'task-event-id',
  taskEvent: {
    id: 'task-event-id',
    taskId: task.id,
    task,
    userId: currentUser.id,
    eventUserId: currentUser.id,
    eventUser: currentUser,
    user: currentUser,
    eventType: 'create_task' as any,
    eventCommentId: null,
    eventComment: null,
    createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
    updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
    deletedAt: null,
  },
  task,
  seenAt: null,
  emailSentAt: null,
  deliveredAt: null,
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
};

export const screeningTool = {
  id: 'screening-tool-id',
  title: 'Screening Tool',
  riskAreaId: riskArea.id,
  riskArea,
  screeningToolScoreRanges: [],
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  patientScreeningToolSubmissions: [],
};

export const screeningToolScoreRange = {
  id: 'screening-tool-score-range-id',
  screeningToolId: screeningTool.id,
  description: 'Screening Tool Score Range',
  minimumScore: 0,
  maximumScore: 10,
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  concernSuggestions: [],
  goalSuggestions: [],
  riskAdjustmentType: 'increment' as any,
};

export const patientScreeningToolSubmission = {
  id: 'patient-screening-tool-submission-id',
  screeningToolId: screeningTool.id,
  patientId: patient.id,
  patient,
  userId: user.id,
  user,
  score: 5,
  scoredAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  deletedAt: null,
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  updatedAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  carePlanSuggestions: [carePlanSuggestionWithConcern, carePlanSuggestionWithGoal],
  screeningToolScoreRangeId: screeningToolScoreRange.id,
  screeningToolScoreRange,
};

export const shortPatientScreeningToolSubmission = {
  id: 'nymeria',
  createdAt: 'Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)',
  score: 5,
  screeningTool: {
    id: screeningTool.id,
    title: screeningTool.title,
    riskAreaId: riskArea.id,
  },
  user: {
    id: user.id,
    firstName: user.firstName,
    lastName: user.lastName,
  },
  screeningToolScoreRange: {
    id: screeningToolScoreRange.id,
    description: screeningToolScoreRange.description,
    riskAdjustmentType: screeningToolScoreRange.riskAdjustmentType,
  },
};

export const shortPatientScreeningToolSubmission2 = {
  id: 'ghost',
  createdAt: 'Thu Jul 11 2017 16:52:56 GMT-0400 (EDT)',
  score: 11,
  screeningTool: {
    id: screeningTool.id,
    title: screeningTool.title,
    riskAreaId: riskArea.id,
  },
  user: {
    id: user.id,
    firstName: user.firstName,
    lastName: user.lastName,
  },
  screeningToolScoreRange: {
    id: screeningToolScoreRange.id,
    description: screeningToolScoreRange.description,
    riskAdjustmentType: screeningToolScoreRange.riskAdjustmentType,
  },
};

export const progressNoteTemplate = {
  id: 'progress-note-template-id',
  title: 'Progress Note Template Title',
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  deletedAt: null,
};

export const progressNote = {
  id: 'progress-note-id',
  createdAt: '2017-09-07T13:45:14.532Z',
  updatedAt: '2017-09-07T13:45:14.532Z',
  startedAt: null,
  location: null,
  deletedAt: null,
  completedAt: null,
  patientId: patient.id,
  summary: 'summary!',
  memberConcern: 'member concern',
  patient,
  user: currentUser,
  progressNoteTemplateId: progressNoteTemplate.id,
  progressNoteTemplate,
  supervisorId: null,
  supervisor: null,
  needsSupervisorReview: null,
  reviewedBySupervisorAt: null,
  supervisorNotes: null,
};

export const quickCall = {
  id: 'quick-call-id',
  createdAt: '2017-11-07T13:45:14.532Z',
  updatedAt: '2017-11-07T13:45:14.532Z',
  deletedAt: null,
  reason: 'Quick Call reason',
  summary: 'Quick Call summary',
  direction: 'Outbound',
  callRecipient: 'Social worker',
  wasSuccessful: true,
  startTime: '2017-11-07T13:45:14.532Z',
  progressNoteId: progressNote.id,
  userId: user.id,
};

export const computedField = {
  id: 'computed-field-id',
  label: 'Computed Field',
  slug: 'computed-field',
  dataType: 'boolean' as any,
  createdAt: '2017-11-07T13:45:14.532Z',
  updatedAt: '2017-11-07T13:45:14.532Z',
  deletedAt: null,
};

export const patientList = {
  id: 'whiteWalkers',
  title: 'White Walkers',
  answerId: 'has-blue-eyes',
  order: 1,
  createdAt: '2017-11-07T13:45:14.532Z',
};

export const patientList2 = {
  id: 'nightsWatch',
  title: 'Protectors of the Seven Kingdoms',
  answerId: 'has-taken-vows',
  order: 2,
  createdAt: '2017-11-07T13:45:14.532Z',
};

export const CBOCategory = {
  id: 'foodServices',
  title: 'Food Services',
};

export const CBOCategory2 = {
  id: 'mentalHealth',
  title: 'Mental Health',
};

export const CBO = {
  id: 'aGirlHasNoPies',
  name: 'Arya Stark Pie Pantry',
  categoryId: CBOCategory.id,
  category: CBOCategory,
  address: 'The Twins',
  city: 'Riverlands',
  state: 'WS',
  zip: '11111',
  phone: '(333) 555-5555',
  fax: '(444) 666-6666',
  url: 'www.facelessmen.com',
  createdAt: '2017-11-07T13:45:14.532Z',
};

export const CBO2 = {
  id: 'evilLaugh',
  name: 'Ramsay Bolton Immersive Therapy',
  categoryId: CBOCategory2.id,
  category: CBOCategory2,
  address: 'Winterfell',
  city: 'The North',
  state: 'WS',
  zip: '12211',
  phone: '(111) 555-5555',
  fax: '(222) 666-6666',
  url: 'www.housebolton.com',
  createdAt: '2017-11-07T13:45:14.532Z',
};

export const CBOReferral = {
  id: 'CBOReferral',
  categoryId: CBOCategory.id,
  category: CBOCategory,
  CBOId: CBO.id,
  CBO,
  name: null,
  url: null,
  diagnosis: null,
  sentAt: null,
  acknowledgedAt: null,
};

export const CBOReferralOther = {
  id: 'CBOReferral',
  categoryId: CBOCategory.id,
  category: CBOCategory,
  CBOId: null,
  CBO: null,
  name: 'Greyjoy Water Therapy',
  url: 'www.thesilence.com',
  diagnosis: null,
  sentAt: null,
  acknowledgedAt: null,
};

export const CBOReferralRequiringAction = {
  id: 'wallHasComeDown',
  categoryId: CBOCategory.id,
  category: CBOCategory,
  CBOId: null,
  CBO: null,
  name: "Night's Watch",
  url: null,
  diagnosis: null,
  sentAt: null,
  acknowledgedAt: null,
};

export const taskWithComment = {
  ...task,
  patient,
  comments: [comment],
  patientId: patient.id,
  assignedTo: currentUser,
  createdBy: currentUser,
  followers: [currentUser],
  patientGoal,
  CBOReferral,
  CBOReferralId: CBOReferral.id,
};

export const basicInfo = {
  patientId: patient.id,
  patientInfoId: patient.patientInfo.id,
  gender: patient.patientInfo.gender,
  language: patient.patientInfo.language,
  primaryAddress: patient.patientInfo.primaryAddress,
  addresses: [address2, address3],
};

export const coreIdentity = {
  firstName: patient.firstName,
  middleName: patient.middleName,
  lastName: patient.lastName,
  dateOfBirth: patient.dateOfBirth,
  patientDataFlags: patient.patientDataFlags,
};

export const completeComputedPatientStatus = {
  id: 'computedPatientStatusId',
  patientId: 'patientId',
  updatedById: 'userId',
  isCoreIdentityVerified: true,
  isDemographicInfoUpdated: true,
  isEmergencyContactAdded: true,
  isAdvancedDirectivesAdded: true,
  isConsentSigned: true,
  isPhotoAddedOrDeclined: true,
  isIneligible: false,
  isDisenrolled: false,
  createdAt: '2017-11-07T13:45:14.532Z',
  updatedAt: '2017-11-07T13:45:14.532Z',
  deletedAt: null,
};

export const incompleteComputedPatientStatus = {
  id: 'computedPatientStatusId',
  patientId: 'patientId',
  updatedById: 'userId',
  isCoreIdentityVerified: true,
  isDemographicInfoUpdated: true,
  isEmergencyContactAdded: true,
  isAdvancedDirectivesAdded: true,
  isConsentSigned: false,
  isPhotoAddedOrDeclined: false,
  isIneligible: false,
  isDisenrolled: false,
  createdAt: '2017-11-07T13:45:14.532Z',
  updatedAt: '2017-11-07T13:45:14.532Z',
  deletedAt: null,
};
