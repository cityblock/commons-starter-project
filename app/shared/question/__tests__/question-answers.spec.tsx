import { shallow } from 'enzyme';

import { clone } from 'lodash';
import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { ENGLISH_TRANSLATION } from '../../../reducers/messages/en';
import ReduxConnectedIntlProvider from '../../../redux-connected-intl-provider';
import { question } from '../../util/test-data';
import QuestionAnswers, { QuestionAnswers as Component } from '../question-answers';

const locale = { messages: ENGLISH_TRANSLATION.messages };
const mockStore = configureMockStore([]);
const answerData = [{ id: 'answser-id', value: '1' }];

it('renders question answers', () => {
  const tree = create(
    <MockedProvider mocks={[]}>
      <Provider store={mockStore({ locale })}>
        <ReduxConnectedIntlProvider>
          <BrowserRouter>
            <QuestionAnswers
              question={question}
              onChange={() => true}
              answerData={answerData}
              editable={true}
            />
          </BrowserRouter>
        </ReduxConnectedIntlProvider>
      </Provider>
    </MockedProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('rendering the correct answer type', () => {
  it('renders radio answers', async () => {
    const component = shallow(
      <Component
        question={question}
        onChange={() => true}
        answerData={answerData}
        editable={true}
      />,
    );
    const instance = component.instance() as Component;
    const result = instance.render();
    expect(result.type.toString()).toMatch(/RadioAnswer/);
  });

  it('renders dropdown answers', async () => {
    const multiSelectQuestion = clone(question);
    multiSelectQuestion.answerType = 'dropdown';

    const component = shallow(
      <Component
        question={multiSelectQuestion}
        onChange={() => true}
        answerData={answerData}
        editable={true}
      />,
    );
    const instance = component.instance() as Component;
    const result = instance.render();
    expect(result.type.toString()).toMatch(/DropdownAnswer/);
  });

  it('renders free text answers', async () => {
    const multiSelectQuestion = clone(question);
    multiSelectQuestion.answerType = 'freetext';

    const component = shallow(
      <Component
        question={multiSelectQuestion}
        onChange={() => true}
        answerData={answerData}
        editable={true}
      />,
    );
    const instance = component.instance() as Component;
    const result = instance.render();
    expect(result.type.toString()).toMatch(/FreeTextAnswer/);
  });

  it('renders multi select answers', async () => {
    const multiSelectQuestion = clone(question);
    multiSelectQuestion.answerType = 'multiselect';

    const component = shallow(
      <Component
        question={multiSelectQuestion}
        onChange={() => true}
        answerData={answerData}
        editable={true}
      />,
    );
    const instance = component.instance() as Component;
    const result = instance.render();
    // This one is weird becuase the top level is not a MultiSelectAnswer
    expect(result.type.toString()).not.toMatch(/RadioAnswer/);
    expect(result.type.toString()).not.toMatch(/FreeTextAnswer/);
    expect(result.type.toString()).not.toMatch(/DropdownAnswer/);
  });
});
