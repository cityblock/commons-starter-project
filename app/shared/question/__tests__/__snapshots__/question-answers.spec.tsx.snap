// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders question answers 1`] = `
<form
  className="container"
>
  <div
    className="container enabled"
  >
    <input
      checked={false}
      disabled={false}
      id="answer-id"
      name={undefined}
      onChange={[Function]}
      onClick={undefined}
      type="radio"
      value="answer-id"
    />
    <label
      htmlFor="answer-id"
    >
      answer value
    </label>
  </div>
</form>
`;
